### DISCUSSION TITLE
Scan feature?

### DISCUSSION THREAD

**Discussion by mgracia9** [svenreiche/Genesis-1.3-Version4#90] [Q&A] [answered]
Is the Scan feature enabled in Genesis1.3 v4? I want to run a scan on the wavelength but I don't really know how to do it in v4, I am setting time=false, it seems it runs a scan but not sure how to set over what variable the scan should be done.

**Comment by svenreiche** [comment, marked as answer]
Hi,
you are on the right track with setting time=false.
Scan is very similar to time-dependent runs except it does not pushes the radiation forward. So as such it is a collection of steady state runs. To set-up scan, e.g. changing the beam emittance the following code should work:

Assume the central wavelength is 1 nm and one would like to do 21 steps in the scan.
The first is the time-frame/scan-frame which is then 21*1nm = 21e-9

&time
slen = 21e-9
sample=1
time=false
&end

Next is defining the scan values, e.g. with profile_polynom. This is the case for emittance

&profile_polynomal
label = scan
c0 = 400e-9
c1 = 5
&end

And define the emittance values with that in the beam namelist

&beam
ex = @scan
ey = @scan
&end

This will produce a scan where the emittance values are varied for each slice form 400 nm to about 510nm

This functionality works for the radiation and beam parameters.  
For the central wavelength and the lattice it is currently not possible since these parameters are used globally in the code (e.g. has only one instance per node and thus it is hard to allow variation between slices.   
For that case a scan in wavelength is (almost) equivalent to a scan in energy (d omega/omega = -2 d gamma/gamma)

I hope this helps

Sven
